src/wallet/wallet.cpp

```diff
@@ -2169,8 +2169,8 @@ std::map<CTxDestination, std::vector<COutput>> CWallet::ListCoins(interfaces::Ch
     std::vector<COutPoint> lockedCoins;
     ListLockedCoins(lockedCoins);
     // Include watch-only for wallets without private keys
-    const bool include_watch_only = IsWalletFlagSet(WALLET_FLAG_DISABLE_PRIVATE_KEYS);
-    const isminetype is_mine_filter = include_watch_only ? ISMINE_WATCH_ONLY : ISMINE_SPENDABLE;
+    const bool include_watch_only = GetLegacyScriptPubKeyMan() && IsWalletFlagSet(WALLET_FLAG_DISABLE_PRIVATE_KEYS);
+    const isminetype is_mine_filter = include_watch_only ? ISMINE_ALL : ISMINE_SPENDABLE;
```

Comment:
Note to other reviewers: `ISMINE_ALL = ISMINE_WATCH_ONLY | ISMINE_SPENDABLE`. When `WALLET_FLAG_DISABLE_PRIVATE_KEYS` is set, no coins can be `ISMINE_SPENDABLE`. Therefore `ISMINE_ALL` leads to the same behavior as `ISMINE_WATCH_ONLY `